synapse/storage/data_stores/main/e2e_room_keys.py

```diff
@@ -236,10 +316,12 @@ def _get_e2e_room_keys_version_info_txn(txn):
                 txn,
                 table="e2e_room_keys_versions",
                 keyvalues={"user_id": user_id, "version": this_version, "deleted": 0},
-                retcols=("version", "algorithm", "auth_data"),
+                retcols=("version", "algorithm", "auth_data", "etag"),
             )
             result["auth_data"] = json.loads(result["auth_data"])
             result["version"] = str(result["version"])
+            if not result["etag"]:
```

Comment:
an explict `if result["etag"] is None` would make the intention here clearer, though in practice it makes little difference.